Modules/Tokens/src/Tokens.Jobs.SanityCheck/Infrastructure/DataSource/DataSource.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Backbone.Modules.Tokens.Application.Infrastructure;
+using Backbone.Modules.Tokens.Domain.Entities;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.BlobStorage;
+
+namespace Tokens.Jobs.SanityCheck.Infrastructure.DataSource
+{
+    public class DataSource : IDataSource
+    {
+        private readonly IBlobStorage _blobStorage;
+        private readonly ITokenRepository _tokenRepository;
+
+        public DataSource(IBlobStorage blobStorage, ITokenRepository tokenRepository)
+        {
+            _blobStorage = blobStorage;
+            _tokenRepository = tokenRepository;
+        }
+
+        public async Task<IEnumerable<string>> GetBlobIdsAsync(CancellationToken cancellationToken)
+        {
+            var blobIds = await _blobStorage.FindAllAsync();
+            return await blobIds.ToListAsync(cancellationToken);
+        }
+
+        public async Task<IEnumerable<TokenId>> GetDatabaseIdsAsync(CancellationToken cancellationToken)
+        {
+            return await _tokenRepository.GetAllTokenIds();
```

Comment:
Oh, the actual reason for asking for the change was, that I wanted you to check the content of expired tokens as well. :)
So you should pass true here. Or to be more expressive: pass `includeExpired: true`)